src/core/frames/frame_controller.ts

```diff
@@ -566,6 +569,7 @@ export class FrameController
   }
 
   private withCurrentNavigationElement(element: Element, callback: () => void) {
+    this.pageSnapshot = PageSnapshot.fromElement(element).clone()
```

Comment:
@manuelpuyol should we have a corresponding `delete this.pageSnapshot`?

```diff
diff --git a/src/core/frames/frame_controller.ts b/src/core/frames/frame_controller.ts
index bc92544..26c01db 100644
--- a/src/core/frames/frame_controller.ts
+++ b/src/core/frames/frame_controller.ts
@@ -553,6 +553,7 @@ export class FrameController
     this.currentNavigationElement = element
     callback()
     delete this.currentNavigationElement
+    delete this.pageSnapshot
   }
 }
 ```

Reply:
Hmm, I tested that idea out locally and tests fail. I can investigate that in the future.

In the meantime, I think cleaning up the assignment would maintain the symmetry in the `withCurrentNavigationElement` implementation.

Reply:
from what I investigated, this is the last call before the frame `src` changes, that's why I added it there
In this case `callback()` is going to change the src and we don't want to cache it with the updated attribute

Reply:
about the `delete`, the problem would be that the `delete` would run before `proposeVisitIfNavigatedWithAction`, which uses the pageSnapshot to pass it forward to the visit

Reply:
we could probably move the snapshot assignment to `navigateFrame`

Reply:
done!